Add some notes to zoneless docs to address some issues (#59689)
angular/angular: adev/src/content/guide/zoneless.md

The `OnPush` change detection strategy is not required, but it is a recommended step towards zoneless compatibility for application components. It is not always possible for library components to use `ChangeDetectionStrategy.OnPush`.
When a library component is a host for user-components which might use `ChangeDetectionStrategy.Default`, it cannot use `OnPush` because that would prevent the child component from being refreshed if it is not `OnPush` compatible and relies on ZoneJS to trigger change detection. Components can use the `Default` strategy as long as they notify Angular when change detection needs to run (calling `markForCheck`, using signals, `AsyncPipe`, etc.).

> Note that being a host for a user component means using an API such as `ViewContainerRef.createComponent` and not simply hosting a portion of a template from a user component (i.e. content projection or a using a template ref input).

Reviewer: nit: avoid "judgement" words like "simply", "basically", etc.

> Note that being a host for a user component means using an API such as `ViewContainerRef.createComponent` and not simply hosting a portion of a template from a user component (i.e. content projection or a using a template ref input).

Reviewer: nit: I try to avoid starting sentences with "Note that", "Notice that", "Be aware that", etc., and just make the statement with preamble.

If you are using SSR with Angular, you may know that it relies on ZoneJS to help determine when the application
is "stable" and can be serialized. If there are asynchronous tasks that should prevent serialization, an application

> not using ZoneJS will need to make Angular aware of these with the [PendingTasks](/api/core/PendingTasks) service. Serialization

Reviewer: ```suggestion
not using ZoneJS must make Angular aware of these with the [PendingTasks](/api/core/PendingTasks) service. Serialization
```
nit: use present tense (over future tense)

will wait for the first moment that all pending tasks have been removed.

The two most straightforward uses of pending tasks are the `run` method:

```typescript
const taskService = inject(PendingTasks);
taskService.run(async () => {
  const someResult = await doSomeWorkThatNeedsToBeRendered();
  this.someState.set(someResult);
});
```

> For more complicated use-cases, manually adding and removing a pending task can be done as follows:

Reviewer: ```suggestion
For more complicated use-cases, you can manually add and remove pending tasks:
```
nit: passive voice

are happening and allow Angular to handle when to synchronize
state rather than manually forcing it to happen in the test.

> The use of `fixture.detectChanges()` is a common pattern in many test

Reviewer 2: This addition is almost immediately contradicting the one above. Perhaps we should combine the two, with a structure like:

> In new code, prefer `await fixture.whenStable()` and avoid using `fixture.detectChanges()`. (rationale)
> When `fixture.detectChanges()` is used in existing code, these are the guarantees: ...

Author: reworded slightly but not entirely. I don't see another reasonable way of doing it.

are happening and allow Angular to handle when to synchronize
state rather than manually forcing it to happen in the test.

For existing test suites, using `fixture.detectChanges()` is a common pattern
and it is likely not worth the effort of converting these to
`await fixture.whenStable()`. `TestBed` will still enforce that the

> fixture's component is `OnPush` compatible and will throw `ExpressionChangedAfterItHasBeenCheckedError`

Reviewer: ```suggestion
fixture's component is `OnPush` compatible and throws `ExpressionChangedAfterItHasBeenCheckedError`
```
